iroh-net/src/discovery.rs

```diff
@@ -95,6 +97,95 @@ pub trait Discovery: std::fmt::Debug + Send + Sync {
     ) -> Option<BoxStream<Result<DiscoveryItem>>> {
         None
     }
+
+    /// Subscribe to all addresses that get discovered.
+    async fn subscribe(&self) -> Option<BoxStream<(NodeId, DiscoveryItem)>> {
```

Comment:
This would be just

```rust
fn subscribe(&self) -> impl Future<Output = Option<BoxStream<(NodeId, DiscoveryItem)>>>
```

I think you can use async even without async_trait now. But the downside is that you can't specify additional bonds like `Send`. If you use the verbose impl Future... syntax you *can* specify additional bonds.